src/router.cpp

```diff
@@ -134,8 +134,12 @@ int zmq::router_t::xsetsockopt (int option_,
 
         case ZMQ_ROUTER_HANDOVER:
             if (is_int && value >= 0) {
-                _handover = (value != 0);
-                return 0;
+                if (!options.connected) {
+                    // We expect nothing is binded yet
+                    _handover = (value != 0);
```

Comment:
Not sure, but if handover is added to options, then maybe it should be only there, and not duplicated as a field of router_t? This opens up to inconsistent states, and may be confusing.

Reply:
True. I can change that, but frankly I do not like this options.handover. We can pass handover flag just in old_pipe->terminate() call. Then receiver will know how to free the session inside session_base_t::pipe_terminated() callback. No incompatibilities.